### PR TITLE
Update to react 16

### PR DESCRIPTION
**Pull request by stereobooster**
Also update all other libraries to the latest versions

### CODE REVIEW

**Comment by matejstrasek** [comment]
Is there any plan when this could be released? :)

**Comment by nassia** [comment]
Also waiting for this.

**Comment by afonsodelgado** [comment]
Any updates on this? Really waiting for it!

**Comment by GiancarlosIO** [comment]
@clauderic Can we merge this pr and release a new version? :pray:

**Comment by GiancarlosIO** [comment]
Any updates? :thinking: :pray:

**Comment by clauderic** [comment]
The scope of this PR is much larger than updating to React 16. Will take a closer look over the weekend.

**Comment by anandrajana** [comment]
@clauderic May I know when can this support react-transition-group v2?. Looking forward to use react-infinite-calendar with react-transition-group v2.

**Comment by kieran** [comment]
would love to see this merged if possible

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 /node_modules
 /umd
 npm-debug.log*
+yarn-error.log
```
Comment (MartinNuc) [inline .gitignore:8]: This shouldnt be here, right?

**Comment by shihabuddin** [comment]
Is this fix released in any version? I am using `2.3.1`. It still has:

```
"peerDependencies": {
  "react": "^15.3.0 || ^16.0.0-alpha"
}
```

**Pull request closed without merging**